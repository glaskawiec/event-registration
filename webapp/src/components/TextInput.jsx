import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'react-bootstrap';
import ValidationBadge from './ValidationBadge';

const TextInput = (props) => (
    <>
        <Form.Control {...props.input} />
        <ValidationBadge
            touched={props.meta.touched}
            error={props.meta.error} />
    </>
)

TextInput.propTypes = {
    input: PropTypes.object.isRequired,
    meta: PropTypes.shape({
        touched: PropTypes.bool,
        error: PropTypes.string,
    }).isRequired
};

export default TextInput;