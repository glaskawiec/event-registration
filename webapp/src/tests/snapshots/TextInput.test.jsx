import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from '../../components/TextInput';

test('TextInput snapshot', () => {
    const props = {
        input: {},
        meta: {
            touched: false,
            error: ''
        }
    };
    const tree = renderer
        .create(
            <TextInput {...props} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});