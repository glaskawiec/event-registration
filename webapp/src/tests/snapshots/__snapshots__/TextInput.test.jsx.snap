// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshot 1`] = `
<input
  className="form-control"
/>
`;
